spec/features/publish/courses/editing_course_outcome_spec.rb

```diff
@@ -0,0 +1,103 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+feature "Editing course outcome" do
+  before do
+    given_i_am_authenticated_as_a_provider_user
+  end
+
+  scenario "i can update the course outcome" do
+    and_there_is_a_qts_course_i_want_to_edit
+    when_i_visit_the_course_outcome_page
+    and_i_update_the_course_outcome
+    and_i_submit
+    then_i_should_see_a_success_message
+    and_the_course_outcome_is_updated
+  end
+
+  scenario "updating with invalid data" do
+    and_there_is_a_course_with_no_outcome
+    when_i_visit_the_course_outcome_page
+    and_i_submit
+    then_i_should_see_an_error_message
+  end
+
+  context "a course offering QTS" do
+    scenario "shows the correct outcome options to choose from" do
+      and_there_is_a_qts_course_i_want_to_edit
+      when_i_visit_the_course_outcome_page
+      then_i_am_shown_the_correct_qts_options
+    end
+  end
+
+  context "a further education course not offering QTS" do
+    scenario "shows the correct outcome options to choose from" do
+      and_there_is_a_non_qts_course_i_want_to_edit
+      when_i_visit_the_course_outcome_page
+      then_i_am_shown_the_correct_non_qts_options
+    end
+  end
+
+  def given_i_am_authenticated_as_a_provider_user
+    given_i_am_authenticated(user: create(:user, :with_provider))
+  end
+
+  def and_there_is_a_qts_course_i_want_to_edit
+    given_a_course_exists(:resulting_in_qts)
+  end
+
+  def and_there_is_a_non_qts_course_i_want_to_edit
+    given_a_course_exists(:resulting_in_pgde, level: "further_education")
+  end
+
+  def and_there_is_a_course_with_no_outcome
+    given_a_course_exists
+    # There is a NOT NULL constraint on the outcome column, so we need to set it
+    # to an out of range enum integer which will force a validation error
+    @course.update_columns(qualification: 99)
```

Comment:
I think this **emphasis** that this simply cannot be a valid test.

We should just remove it otherwise we testing the **_impossible_** simply because it is **_possible_**.

Reply:
yeah i was wondering whether it was worth testing this. Agreed, will remove

Reply:
removed